File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Core/Models/TaskState.swift
import Foundation

/// Enum состояния задачи на основе спецификации Memory Bank
/// Разрешенные переходы:
/// - NEW → IN_PROGRESS
/// - IN_PROGRESS → PAUSED
/// - PAUSED → IN_PROGRESS
/// - IN_PROGRESS → COMPLETED
enum TaskState: String, Codable, CaseIterable {
    case new = "NEW"
    case inProgress = "IN_PROGRESS"
    case paused = "PAUSED"
    case completed = "COMPLETED"

    var displayName: String {
        switch self {
        case .new:
            return "Новая"
        case .inProgress:
            return "В работе"
        case .paused:
            return "Приостановлена"
        case .completed:
            return "Завершена"
        }
    }

    var color: String {
        switch self {
        case .new:
            return "blue"
        case .inProgress:
            return "green"
        case .paused:
            return "orange"
        case .completed:
            return "gray"
        }
    }

    /// Возвращает разрешенные следующие состояния на основе текущего состояния
    var allowedTransitions: [TaskState] {
        switch self {
        case .new:
            return [.inProgress]
        case .inProgress:
            return [.paused, .completed]
        case .paused:
            return [.inProgress]
        case .completed:
            return []
        }
    }

    /// Проверяет, допустим ли переход к другому состоянию
    func canTransition(to newState: TaskState) -> Bool {
        return allowedTransitions.contains(newState)
    }
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/core/models/TaskState.kt
package com.beyondviolet.wfm.core.models

import androidx.compose.ui.graphics.Color
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Enum состояния задачи на основе спецификации Memory Bank
 * Разрешенные переходы:
 * - NEW → IN_PROGRESS
 * - IN_PROGRESS → PAUSED
 * - PAUSED → IN_PROGRESS
 * - IN_PROGRESS → COMPLETED
 */
@Serializable
enum class TaskState {
    @SerialName("NEW")
    NEW,

    @SerialName("IN_PROGRESS")
    IN_PROGRESS,

    @SerialName("PAUSED")
    PAUSED,

    @SerialName("COMPLETED")
    COMPLETED;

    val displayName: String
        get() = when (this) {
            NEW -> "Новая"
            IN_PROGRESS -> "В работе"
            PAUSED -> "Приостановлена"
            COMPLETED -> "Завершена"
        }

    val color: Color
        get() = when (this) {
            NEW -> Color(0xFF2196F3) // Blue
            IN_PROGRESS -> Color(0xFF4CAF50) // Green
            PAUSED -> Color(0xFFFF9800) // Orange
            COMPLETED -> Color(0xFF9E9E9E) // Gray
        }

    /**
     * Возвращает разрешенные следующие состояния на основе текущего состояния
     */
    val allowedTransitions: List<TaskState>
        get() = when (this) {
            NEW -> listOf(IN_PROGRESS)
            IN_PROGRESS -> listOf(PAUSED, COMPLETED)
            PAUSED -> listOf(IN_PROGRESS)
            COMPLETED -> emptyList()
        }

    /**
     * Проверяет, допустим ли переход к другому состоянию
     */
    fun canTransitionTo(newState: TaskState): Boolean {
        return allowedTransitions.contains(newState)
    }
}
